File: GeeksforGeeks/Python3/Medium/Bitwise AND of the Array/bitwise-and-of-the-array.py
#User function Template for python3
class Solution:
    def count(self, N, A, X): 
        # code here
        prefix = 0
        ans = N
        for i in range(30, -1, -1):
            if X & (1 << i):
                prefix |= 1 << i
                continue
            cnt = 0
            prefix |= 1 << i
            for a in A:
                if (a & prefix) == prefix:
                    cnt += 1
            ans = min(ans, N - cnt)
            prefix ^= 1 << i
        return ans

#{ 
 # Driver Code Starts
#Initial Template for Python 3




if __name__ == '__main__': 
    t = int(input())
    for _ in range(t):
        N,X = map(int,input().strip().split())
        A = list(map(int, input().strip().split()))
        ob = Solution()
        ans = ob.count(N, A, X)
        print(ans)
# } Driver Code Ends

File: GeeksforGeeks/C++/Medium/Bitwise AND of the Array/bitwise-and-of-the-array.cpp
//{ Driver Code Starts
//Initial Template for C++
#include<bits/stdc++.h>
using namespace std;

// } Driver Code Ends
//User function Template for C++
class Solution{
    public:
    int count(int N, vector<int> A,int X)
    {
        // code here
        int temp=0,ans=N;
        for(int i=31;i>=0;i--){
            if((X>>i)&1){
                temp|=(1<<i);
            }
            else{
                int p=temp|(1<<i);
                int cnt=0;
                for(auto j:A){
                    if((j&p)==p)cnt++;
                }
            ans=min(ans,N-cnt);
            }
        }
        return ans;
    }
};

//{ Driver Code Starts.
int main() {
    int t;
    cin>>t;
    while(t--)
    {
        int N,X;
        cin>>N>>X;
        vector<int> A(N);
        for(auto &i:A)
            cin>>i;
        Solution obj;
        int ans = obj.count(N, A, X);
        cout<<ans<<endl;
    }
}
// } Driver Code Ends